packages/edit-navigation/src/components/menu-locations-editor/index.js

```diff
@@ -1,3 +1,117 @@
+/**
+ * External dependencies
+ */
+import { forEach, find, pick } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	Panel,
+	PanelBody,
+	SelectControl,
+	Spinner,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const MenuSelectControl = ( { onChange, menus, menusList, location } ) => {
+	const [ menuId, setMenuId ] = useState( false );
+
+	return (
+		<>
+			{ menus && (
+				<SelectControl
+					options={ menusList }
+					value={ menuId || location.menu }
+					onChange={ ( selectedMenuId ) => {
+						setMenuId( selectedMenuId );
+						onChange( {
+							location: location.name,
```

Comment:
I'd suggest calling this `locationName` instead of `location`.

Reply:
I would leave it like this b/c it plays nice with the API which expects `location` as name